src/Query/Providers/QueryProvider.cs

```diff
@@ -344,7 +344,23 @@ public int ExecuteNonQuery(string query, CommandParameterCollection parameters)
         }
 
         /// <inheritdoc/>
-        public async Task<int> ExecuteNonQueryAsync(string query) => await ExecuteNonQueryAsync(query, null);
+        public async Task<int> ExecuteNonQueryAsync(string query)
+            => await ExecuteNonQueryAsync(query, new CommandParameterCollection());
+
+        /// <inheritdoc/>
+        public Task<int> ExecuteNonQueryAsync(string query, params object[] parameters)
+        {
+            var paramsCollection = new CommandParameterCollection();
+            var tempParameters = new ParamEnumerator(query);
+
+            foreach (var parameter in parameters)
```

Comment:
We should probably check if the number of parameters in query is the same as the number of values and throw reasonable exception if not.

- If we are still inside `foreach`, `tempParameters.MoveNext()` should return `true`. Now if there are more values than parameters in query, the last value will be used for all remaining parameters.
- If we are after `foreach`, `tempParameters.MoveNext()` should return `false`. If not, it means that there are more query parameters than values and so some parameters are not set.